applications/GeoMechanicsApplication/custom_strategies/strategies/README.md

```diff
@@ -0,0 +1,93 @@
+growth is finished or all piping elements are checked.
+
+### FinalizeSolutionStep function
+
+![FinalizeSolutionStep.svg](FinalizeSolutionStep.svg)
+
+This function calls other functions that are described below.
+
+### GetPipingElements function
+
+This function returns a list of all piping elements. To create the list, all elements are checked for  
+PIPE_START_ELEMENT property that holds Id of the piping starting element.
+Then elements on the list are sorted based on X-coordinate of the element begin.
+The current limitation is the following. The starting piping element has to have minimum or maximum value of the X
+coordinate. This is needed to find a direction for the piping growth.
+
+### InitialiseNumActivePipeElements function
+
+It checks the list of the elements and returns a number of elements with PIPE_ACTIVE property At the beginning the
+number is zero.
+
+### CalculateMaxPipeHeight function
+
+It calculates a maximum diameter using CalculateParticleDiameter function then multiplies it by a magic factor of 100 to
+get the height value.
+
+### CalculateParticleDiameter
+
+In case of PIPE_MODIFIED_D, the function returns
+
+$$ diameter = 2.08\times 10^{-4} * \Bigg(\frac{D_{70}}{2.08\times 10^{-4}}\Bigg)^{0.4}$$
+
+otherwise, $$diameter = D_{70}$$
+
+### check_pipe_equilibrium function
+
+This function performs iterative process with a prescribed maximum number of iterations. For each iteration the
+modelling is
+performed. Then a pipe height is calculated for each open piping element.
+
+![check_pipe_equilibrium.svg](check_pipe_equilibrium.svg)
+
+### Recalculate function
+
+This function performs the modeling using prescribed GeoMechanicsNewtonRaphsonStrategy class.
+
+![Recalculate.svg](Recalculate.svg)
+
+### CalculateEquilibriumPipeHeight function
+
+This is a function from SteadyStatePwPipingElement. It calculates the equilibrium pipe height as
+
+$$F \frac{\pi }{3} D_p \Big(\frac{\rho_s}{\rho_w} - 1\Big) \eta
+\frac{\sin(\theta + \alpha)}{\cos(\theta)  \frac{dh}{dx}}$$
+
+where $F$ is the model factor,\
+$D_p$ is the particle diameter calculated with CalculateParticleDiameter function,\
+$\rho_s$ is the solid density,\
+$\rho_w$ is the water density,\
+$\eta=\frac{n d^2}{A}$ is the packing coefficient?\
+$\theta$ is the angle of repose or bedding angle?\
+$\alpha$ is the slope of the pipe,\
+$\frac{dh}{dx}$ is the head gradient obtained with CalculateHeadGradient function,\
+Currently, the pipe is assumed horizontal $\alpha = 0$.
+
+### check_status_tip_element function
+
+This function checks the open piping elements and if their height is larger than a maximum value of the pipe height or
+too small then the function changes the pipe properties: PIPE_EROSION and PIPE_ACTIVE to false.
+
+![check_status_tip_element.svg](check_status_tip_element.svg)
+
+### save_or_reset_pipe_heights function
+
+Based on the grow value, this function sets PREV_PIPE_HEIGHT equal to PIPE_HEIGHT or reset PIPE_HEIGHT to
```

Comment:
To me it's not yet completely clear what 'grow' means. Does it mean the element is still allowed to grow?

Reply:
`grow` is an input, this function just sets values from one place to another.

Reply:
Indeed, but what does the input mean?

Reply:
When grow is true, the piping is growing. The grow value is determined in the check_status_tip_element function. Info on grow has been added now.